Move substituteParams into its own module

valadaptive opened the pull request:
This is a first step towards a macro system refactor.

I've decided not to import name1 and name2 directly into the new module, but instead pass them in to all substituteParams calls.

I have no way of knowing whether name1 and name2 are actually useful in any given context where substituteParams is called, but this leaves the door open to future refactors without breaking anything, and by making it explicit, this should help people think about whether they *really* want to pass name1 and name2 in.

If you're up to it, it may be worth going through some of the substituteParams calls now, and removing name1 and name2 from those which don't need them.

Cohee1207 commented:
This will break downloadable extensions and third-party extensions that we're not aware of.
I suggest that you leave the old substituteParams function as a proxy and wrap the call into a new function that will accept name1, name2, and other parameters, maybe even as an object to simplify future additions and refactorings.

valadaptive committed:
Move macro substitution to new module

substituteParams has become a thin wrapper around the new evaluateMacros
function, and will become more of a compatibility shim as refactorings
and rewrites are done.

valadaptive commented:
That approach is actually way nicer--I'll be able to turn `substituteParams` into a compatibility shim for the new macro evaluation API in whatever form it takes.

Cohee1207 commented on public/script.js:
```diff
@@ -2146,187 +2061,9 @@ function getCurrentSwipeId() {
-    content = bannedWordsReplace(content);
-    content = randomReplace(content);
-    return content;
+    return evaluateMacros(content, _name1 ?? name1, _name2 ?? name2, _original, _group ?? name2);
```
`_replaceCharacterCard` param is not passed down to the shim. Is that correct?

valadaptive replied:
Good catch--I missed it because it's not in the JSDoc

Cohee1207 committed:
Fix relative imports

Cohee1207 committed:
Add missing arg to JSdoc
